rotkehlchen/chain/ethereum/modules/thegraph/balances.py

```diff
@@ -0,0 +1,122 @@
+        balances: BalancesType = defaultdict(lambda: defaultdict(Balance))
+        grt_price = Inquirer().find_usd_price(A_GRT)
+
+        # fetch deposit events
+        addresses_with_deposits = self.addresses_with_deposits(
+            product=EvmProduct.STAKING,
+            deposit_events=self.get_deposit_events(),
+        )
+        # remap all events into a list that will contain all pairs (delegator, indexer)
+        delegations = set()
+        for delegator, event_list in addresses_with_deposits.items():
+            for event in event_list:
+                if event.extra_data is None:
+                    continue
+                if (indexer := event.extra_data.get('indexer')) is not None:
+                    delegations.add((delegator, indexer))
+        delegations_unique = list(delegations)
+
+        staking_contract = self.evm_inquirer.contracts.contract(self.staking_contract)
+        chunk_size, call_order = get_chunk_size_call_order(self.evm_inquirer)
+
+        # query how many shares delegator currently have at the indexer/pool
+        all_shares = self.evm_inquirer.multicall(
+            calls=[(
+                staking_contract.address,
+                staking_contract.encode(
+                    method_name='getDelegation',
+                    arguments=[indexer, delegator],
+                ),
+            ) for delegator, indexer in delegations_unique],
+            call_order=call_order,
+            calls_chunk_size=chunk_size,
+        )
+
+        # query pool data to get current total amount of shares and tokens in that pool
+        pools = self.evm_inquirer.multicall(
+            calls=[(
+                staking_contract.address,
+                staking_contract.encode(
+                    method_name='delegationPools',
+                    arguments=[indexer],
+                ),
+            ) for _, indexer in delegations_unique],
+            call_order=call_order,
+            calls_chunk_size=chunk_size,
+        )
+
+        for idx, delegation in enumerate(delegations_unique):
+            delegator = delegation[0]
+            shares_response = staking_contract.decode(
+                all_shares[idx],
+                'getDelegation',
+                arguments=[delegator, delegation[1]],
+            )[0]
+            pool_response = staking_contract.decode(
+                result=pools[idx],
+                method_name='delegationPools',
+                arguments=[delegator],
+            )[0]
+            shares = shares_response[0]
+            pool_total_tokens = pool_response[4]
+            pool_total_shares = pool_response[5]
+
+            # calculate current tokens balance relative to the pool state and shares distribution
+            if pool_total_shares == 0:
+                continue
+            balance = FVal(shares * pool_total_tokens / pool_total_shares)
+            balance_norm = asset_normalized_value(balance.to_int(exact=False), self.grt)
```

Comment:
do you need this `to_int`? I believe that if the signature is an integer this is already an integer

Reply:
`mypy` complains w/o `to_int()`:
```bash
error: Argument 1 to "asset_normalized_value" has incompatible type "FVal"; expected "int"  [arg-type]
```

Reply:
it's needed due to the signature of `asset_normalized_value` and `token_normalized_value_decimals`.